test/Test/Data/Mutable/HashMap.hs

```diff
@@ -1,16 +1,228 @@
+{-# LANGUAGE LinearTypes #-}
 {-# LANGUAGE OverloadedStrings #-}
-{-# LANGUAGE TemplateHaskell #-}
+{-# OPTIONS_GHC -Wno-name-shadowing #-}
 
-module Test.Data.Mutable.HashMap (mutHMTests) where
+-- |
+-- Tests for mutable hashmaps
+-- We are excluding tests for alter because it's assumed that alter is
+-- implemented parametrically around insert and delete.
+module Test.Data.Mutable.HashMap
+  ( mutHMTests,
+  )
+where
 
+import qualified Data.Functor.Linear as Lin
+import qualified Data.HashMap.Linear as HMap
+import Data.Unrestricted.Linear
 import Hedgehog
+import qualified Hedgehog.Gen as Gen
+import qualified Hedgehog.Range as Range
+import qualified Prelude.Linear as Lin
+import Test.Tasty
+import Test.Tasty.Hedgehog (testProperty)
+import Test.Util
 
-mutHMTests :: IO Bool
-mutHMTests = checkParallel group
+-- # Exported Tests
+--------------------------------------------------------------------------------
 
-group :: Group
-group = Group "Mutable hashmap tests" $
-  []
+mutHMTests :: TestTree
+mutHMTests = testGroup "Mutable hashmap tests" group
 
+group :: [TestTree]
+group =
+  [ testProperty "[singletons work correctly]" testSingleton,
+    testProperty "[lookup after insert]" insertLookup,
```

Comment:
There are two possible tests here: lookup of what you've inserted, and lookup of what you haven't inserted. Both tests are useful.